#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 0de14a576faf93d0a540ce65aa4c5a5a203e35c7 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Tue, 1 Mar 2016 17:41:58 +0100
Subject: [PATCH] Fix Kdelibs4 config migration for Windows

Look for old config in ~/AppData/Roaming/.kde on Windows and
use GenericConfigLocation instead of ConfigLocation for new
config.
---
 src/lib/util/kdelibs4configmigrator.cpp | 2 +-
 src/lib/util/kdelibs4migration.cpp      | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/lib/util/kdelibs4configmigrator.cpp b/src/lib/util/kdelibs4configmigrator.cpp
index c8560c1..acdb19c 100644
--- a/src/lib/util/kdelibs4configmigrator.cpp
+++ b/src/lib/util/kdelibs4configmigrator.cpp
@@ -72,7 +72,7 @@ bool Kdelibs4ConfigMigrator::migrate()

     Q_FOREACH (const QString &configFileName, d->configFiles) {
         const QString newConfigLocation
-            = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation)
+            = QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation)
               + QLatin1Char('/') + configFileName;

         if (QFile(newConfigLocation).exists()) {
diff --git a/src/lib/util/kdelibs4migration.cpp b/src/lib/util/kdelibs4migration.cpp
index f631efa..32a78c0 100644
--- a/src/lib/util/kdelibs4migration.cpp
+++ b/src/lib/util/kdelibs4migration.cpp
@@ -41,6 +41,9 @@ Kdelibs4Migration::Kdelibs4Migration()
         QDir homeDir = QDir::home();
         QVector<QString> testSubdirs;
         testSubdirs << QStringLiteral(KDE4_DEFAULT_HOME) << QStringLiteral(".kde4") << QStringLiteral(".kde");
+#ifdef Q_OS_WIN
+        testSubdirs << QStringLiteral("AppData/Roaming/.kde");
+#endif
         Q_FOREACH (const QString &testSubdir, testSubdirs) {
             if (homeDir.exists(testSubdir)) {
                 //qDebug() << "Using" << testSubdir << "as the location of the old config file";
--
2.1.4
